modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbStorageEngine.java

```diff
@@ -75,19 +78,27 @@ public RocksDbStorageEngine(RocksDbStorageEngineConfiguration engineConfig, Path
     /** {@inheritDoc} */
     @Override
     public void start() throws StorageException {
-        RocksDbDataRegion defaultRegion = new RocksDbDataRegion(engineConfig.defaultRegion());
+        registerDataRegion(engineConfig.defaultRegion());
+
+        // TODO: IGNITE-17066 Add handling deleting/updating data regions configuration
+        engineConfig.regions().listenElements(new ConfigurationNamedListListener<>() {
+            @Override
+            public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<RocksDbDataRegionView> ctx) {
+                registerDataRegion(ctx.config(RocksDbDataRegionConfiguration.class));
 
-        defaultRegion.start();
+                return CompletableFuture.completedFuture(null);
+            }
+        });
+    }
 
-        regions.put(DEFAULT_DATA_REGION_NAME, defaultRegion);
+    private void registerDataRegion(RocksDbDataRegionConfiguration dataRegionConfig) {
+        var region = new RocksDbDataRegion(dataRegionConfig);
 
-        for (String regionName : engineConfig.regions().value().namedListKeys()) {
```

Comment:
This shouldn't be removed, we still need to create preconfigured regions

Reply:
Actually, no. When creating an Ignite instance, listeners get notified on start, please see the `IgniteImpl#notifyConfigurationListeners` method